pages/home.page.ts

```
@@ -12,4 +23,49 @@ export class HomePage {
   async openProduct(productName: string) {
     await this.page.getByText(productName).click();
   }
+
+  async getAllProductsOnPage(): Promise<string[]> {
+    return await this.productsTitle.allTextContents();
+  }
+
+  async getAllProductsPricesOnPage(): Promise<number[]> {
+    const prices = await this.productsPrice.allTextContents();
+    return prices.map((price) => parseFloat(price.replace("$", "")));
+  }
+
+  async verifyProductsFiltered(
+    filterOption: HandToolsCategories | PowerToolsCategories
+  ) {
+    if (!(await this.noProductsResult.isVisible())) {
+      const productsTitle = await this.getAllProductsOnPage();
+      for (const product of productsTitle) {
+        expect(product.trim()).toContain(filterOption);
+      }
+    } else console.log("There are no products found.");
```

it might be better to throw an error in this case?

---

not sure if it is an error if no products found by filtered category. I think it is a  real case